src/commands/PublishCommand.js

```diff
@@ -591,7 +591,14 @@ export default class PublishCommand extends Command {
     });
 
     if (this.gitEnabled) {
-      changedFiles.forEach((file) => GitUtilities.addFile(file, this.execOpts));
+      changedFiles.forEach((file) => {
+        const ignored = GitUtilities.checkIgnore(file, this.execOpts);
```

Comment:
How about instead we just add [`--update`](https://git-scm.com/docs/git-add#git-add---update) to the `git add` command? It's been around for almost ten years, and disallows adding new files (silently ignoring ignored files), which is okay since we're never adding new files anyway when publishing.

I'd rather not do a bunch of extra calls to git when all we need to do is add an extra flag to an existing call.

Reply:
Ah yes I hadn't considered `--update`. I will do it that way instead.

Reply:
I was just browsing through pull requests to see what's coming up and this is the first time I hear of the this amazing git feature. Completely off-topic, I know, but reading this made me a happy(ier) man.